gems/aws-sdk-s3/lib/aws-sdk-s3/customizations/object.rb

```diff
@@ -312,6 +312,8 @@ def upload_stream(options = {}, &block)
       #   multipart uploads. This option is not used if the file is smaller than
       #   `:multipart_threshold`.
       #
+      # @option options [Proc] :progress_callback - TBD
```

Comment:
progress_callback makes sense.. but it's also limiting to what can be done here. would this be only used in that case? if not then probably a new generic name would be better